src/persist-txn/src/txns.rs

```diff
@@ -318,27 +401,6 @@ where
         debug!("apply_le {:?}", ts);
         self.txns_cache.update_gt(ts).await;
 
-        // TODO(txn): Oof, figure out a protocol where we don't have to do this
-        // every time.
-        let inits = FuturesUnordered::new();
-        for (data_id, times) in self.txns_cache.datas.iter() {
-            let register_ts = times.first().expect("data shard has register ts").clone();
-            let mut data_write = self.datas.take_write(data_id).await;
-            inits.push(async move {
-                let () = crate::empty_caa(
-                    || format!("data init {:.9}", data_id.to_string()),
-                    &mut data_write,
-                    register_ts,
-                )
-                .await;
-                data_write
-            });
-        }
-        let data_writes = inits.collect::<Vec<_>>().await;
-        for data_write in data_writes {
-            self.datas.put_write(data_write);
-        }
```

Comment:
Why are we getting rid of this?

Reply:
I realized that we no longer needed it now that we switched to the `unblock_reads` approach for reads. In a sense, I'm doing the TODO :D

This is also related to what I was (poorly) trying to explain in the sync yesterday. Once we add compaction, we lose information about the original registration timestamp (it gets compacted forward), so we'd either have to continually do empty CaAs on all registered data shards to maintain this sort of invariant or figure out how to deal without it (i.e. unblock_reads instead of "translating" the as_of)